packages/design-system/src/components/table/components/filtersSidebar/index.tsx

```diff
@@ -33,19 +33,71 @@ const FiltersSidebar = ({
   filters,
   toggleFilterSelection,
 }: FiltersSidebarProps) => {
-  if (!Object.keys(filters).length) {
+  const [expandAll, setExpandAll] = useState(false);
+  const expandedFilters = useRef(new Set<string>());
+  const filterKeys = useMemo(() => {
+    return Object.keys(filters);
+  }, [filters]);
```

Comment:
These `useMemo` usage look unnecessary to me.

Reply:
This is required as the `filters` object contains `filterValues`, which keep changing as cookies update frequently.
So if cookies update, `filterValues` update, and eventually the `filters` object updates, because of that the entire object should be memoized.